services/motion/builtin/state/state.go

```diff
@@ -0,0 +1,647 @@
+// Package state provides apis for motion builtin plan executions
+// and manages the state of those executions
+package state
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/google/uuid"
+	geo "github.com/kellydunn/golang-geo"
+	"github.com/pkg/errors"
+
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/services/motion"
+	"go.viam.com/rdk/spatialmath"
+)
+
+var (
+	// ErrUnknownResource indicates that the resource is not known.
+	ErrUnknownResource = errors.New("unknown resource")
+	// ErrNotFound indicates the entity was not found.
+	ErrNotFound = errors.New("not found")
+)
+
+// State is the state of the builtin motion service
+// It keeps track of the builtin motion service's executions.
+type State struct {
+	waitGroup  *sync.WaitGroup
+	cancelCtx  context.Context
+	cancelFunc context.CancelFunc
+	logger     golog.Logger
+	// mu protects the componentStateByComponent
+	mu                        sync.RWMutex
+	componentStateByComponent componentStateByComponent
+}
+
+// ExecutionReq is the request type for StartExecution.
+type ExecutionReq struct {
```

Comment:
[opt] As part of https://viam.atlassian.net/browse/RSDK-5146, we'll need to extend this to support MoveOnMap requests. Not sure if you want to think about how to do that now and/or name this so it's clear it's only for MoveOnGlobe.

Reply:
I think this struct should be able to be used for MoveOnMap as well, we will just need to add some new fields to support it & have a constructor for move on map.

I think that work can be done when we start making move on map async.

Does that sound ok?

Reply:
Yes